From: Michael Chan <mchan@broadcom.com>
Date: Sun, 27 Dec 2015 18:19:22 -0500
Subject: [PATCH 05/74] bnxt_en: Allocate rx_cpu_rmap only if Accelerated RFS
 is enabled.
Patch-mainline: v4.5-rc1
Git-commit: 45019a180358c3cf290c3f3dc953c44f978d5527
References: bsc#963575 FATE#320144

Also, no need to check for bp->rx_nr_rings as it is always >= 1.  If the
allocation fails, it is not a fatal error and we can still proceed.

Signed-off-by: Michael Chan <mchan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/broadcom/bnxt/bnxt.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/net/ethernet/broadcom/bnxt/bnxt.c
+++ b/drivers/net/ethernet/broadcom/bnxt/bnxt.c
@@ -4009,10 +4009,8 @@ static int bnxt_set_real_num_queues(stru
 		return rc;
 
 #ifdef CONFIG_RFS_ACCEL
-	if (bp->rx_nr_rings)
+	if (bp->flags & BNXT_FLAG_RFS)
 		dev->rx_cpu_rmap = alloc_irq_cpu_rmap(bp->rx_nr_rings);
-	if (!dev->rx_cpu_rmap)
-		rc = -ENOMEM;
 #endif
 
 	return rc;
